cmd/kube-controller-manager/app/options/options.go

```diff
@@ -295,6 +298,15 @@ func (s *KubeControllerManagerOptions) Flags(allControllers []string, disabledBy
 	fs.StringVar(&s.Master, "master", s.Master, "The address of the Kubernetes API server (overrides any value in kubeconfig).")
 	fs.StringVar(&s.Generic.ClientConnection.Kubeconfig, "kubeconfig", s.Generic.ClientConnection.Kubeconfig, "Path to kubeconfig file with authorization and master location information (the master location can be overridden by the master flag).")
 
+	if !utilfeature.DefaultFeatureGate.Enabled(featuregate.Feature(clientgofeaturegate.WatchListClient)) {
+		if err := utilfeature.DefaultMutableFeatureGate.OverrideDefault(featuregate.Feature(clientgofeaturegate.WatchListClient), true); err != nil {
```

Comment:
this doesn't seem great to leave like this in non-test code ... can we limit this to test code or make the tests behave more correctly?

Reply:
In general setting a default value for KCM makes it visible on the command line to end users. 
`(WatchListClient=true|false (BETA - default=true))`


Initially, this code would panic if setting the default failed. 

Unfortunately, that caused the `TestComponentSecureServingAndAuth` test to fail. The current feature gate implementation relies on global state, and the mentioned test runs more than one component (server and KCM), which triggers the panic. 

We’ve modified the code to log the error instead of panicking. This is a temporary solution until the default value for the feature gate is properly set.